tests/frame/test_invalid.py

```diff
@@ -24,7 +24,7 @@ def test_validate_laziness() -> None:
         NotImplementedError,
         match=("The items to concatenate should either all be eager, or all lazy"),
     ):
-        nw.concat([nw.from_native(df, eager_only=True), nw.from_native(df).lazy()])
+        nw.concat([nw.from_native(df, eager_only=True), nw.from_native(df).lazy()])  # type: ignore[list-item]
```

Comment:
I am arguing that this is correct for mypy to complain, as it is supposed to raise an error and that's what we are checking

Reply:
yup, and this is quite satisfying 😋